syntax = "proto3";

package translator;

option go_package = "translator/proto";

service Translator {
  rpc Translate (TranslateRequest) returns (TranslateResponse) {}
  rpc BatchTranslate (BatchTranslateRequest) returns (BatchTranslateResponse) {}
}

message TranslateRequest {
  string text = 1;
  string source_lang = 2;
  string target_lang = 3;
}

message TranslateResponse {
  string translated_text = 1;
}

message BatchTranslateRequest {
  repeated string texts = 1;
  string source_lang = 2;
  string target_lang = 3;
}

message BatchTranslateResponse {
  repeated string translated_texts = 1;
} 